boa_examples/src/bin/jspromise.rs

```diff
@@ -0,0 +1,23 @@
+use std::error::Error;
+use boa_engine::{
+job::SimpleJobQueue,
+object::builtins::JsPromise,
+builtins::promise::PromiseState,
+Context, JsValue, js_string};
+
+fn main() -> Result<(), Box<dyn Error>> {
```

Comment:
The example looks good, but looking at other examples, we should be adding more comments through the code, explaining what each step does.

Also, is there any other API usage we can show?

Reply:
Added comments to the code. 

Do you mean that this example should use more of the api or that there should be another jspromise example in addition to this one?

Reply:
Probably that the example should use more methods of `JsPromise`.